tests/reporter/kubernetes.go

```diff
@@ -918,3 +1024,157 @@ func writeStringToFile(filePath string, data string) error {
 	_, err = f.WriteString(data)
 	return err
 }
+
+func getVmiType(vmi v12.VirtualMachineInstance) (string, error) {
+	for _, volume := range vmi.Spec.Volumes {
+		if volume.VolumeSource.ContainerDisk == nil {
+			continue
+		}
+
+		image := volume.VolumeSource.ContainerDisk.Image
+		if strings.Contains(image, "fedora") {
+			return "fedora", nil
+		} else if strings.Contains(image, "cirros") {
+			return "cirros", nil
+		} else if strings.Contains(image, "alpine") {
+			return "alpine", nil
+		}
+	}
+
+	return "", fmt.Errorf("unknown type, vmi %s", vmi.ObjectMeta.Name)
```

Comment:
If you choose to return if containerDisk is nil, you can also return the unknown type here:
```
return "", fmt.Errorf("unknown type %s for vmi %s", volume.VolumeSource.ContainerDisk.Image, vmi.ObjectMeta.Name) 
```

Reply:
It might have several ContainerDisk volumes IIUC,
if so i can't print them